frontend/viewer/src/lib/writing-system-runes.svelte.ts

```diff
@@ -0,0 +1,29 @@
+import {untrack} from 'svelte';
+import type {IWritingSystems} from './dotnet-types';
+import {useMiniLcmApi} from './services/service-provider';
+import {initWritingSystemService, WritingSystemService} from './writing-system-service';
+import {writable} from 'svelte/store';
+
+//todo this won't work when projects change, the WSS should depend on miniLcmApi which should also be reactive
+let writingSystems = $state< IWritingSystems | null>(null);
+let loading = $state(false);
+export function useWritingSystemRunes(): WritingSystemService {
+  if (!writingSystems){
+    untrack(load);
+  }
+
+  return new WritingSystemService(writingSystems ?? {analysis: [], vernacular: []});
```

Comment:
This code is super confusing 😬. And `writingSystems` gets assigned to in the resolution of a promise. Does that really happen imediately? Isn't `writingSystems` still undefined here on the first run? And `initWritingSystemService(wsStore);` is essentially for backwards compatibility for the svelte-ux editor?

Reply:
Yes that's correct. It also doesn't always work lol. Basically it starts and works and once it loads this function will get executed again when this state variable changes (inside the promise). The right way to fix this would be to move the state inside the class and execute the query inside the class constructor. Then any code which calls the functions on the service will depend on the writing systems, so if they change then they will also update. But I'm not sure we can use state like that from non rune components.